src/contracts/pods/EigenPod.sol

```diff
@@ -159,143 +113,141 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
+     * (see `_updateCheckpoint` for more details)
+     * @dev This method can only be called when there is a currently-active checkpoint.
+     * @param balanceContainerProof proves the beacon's current balance container root against a checkpoint's `beaconBlockRoot`
+     * @param proofs Proofs for one or more validator current balances against the `balanceContainerRoot`
      */
-    function verifyAndProcessWithdrawals(
-        uint64 oracleTimestamp,
-        BeaconChainProofs.StateRootProof calldata stateRootProof,
-        BeaconChainProofs.WithdrawalProof[] calldata withdrawalProofs,
-        bytes[] calldata validatorFieldsProofs,
-        bytes32[][] calldata validatorFields,
-        bytes32[][] calldata withdrawalFields
-    ) external onlyWhenNotPaused(PAUSED_EIGENPODS_VERIFY_WITHDRAWAL) {
+    function verifyCheckpointProofs(
+        BeaconChainProofs.BalanceContainerProof calldata balanceContainerProof,
+        BeaconChainProofs.BalanceProof[] calldata proofs
+    ) 
+        external 
+        onlyWhenNotPaused(PAUSED_EIGENPODS_VERIFY_CHECKPOINT_PROOFS) 
+    {
+        uint64 checkpointTimestamp = currentCheckpointTimestamp;
         require(
-            (validatorFields.length == validatorFieldsProofs.length) &&
-                (validatorFieldsProofs.length == withdrawalProofs.length) &&
-                (withdrawalProofs.length == withdrawalFields.length),
-            "EigenPod.verifyAndProcessWithdrawals: inputs must be same length"
+            checkpointTimestamp != 0, 
+            "EigenPod.verifyCheckpointProofs: must have active checkpoint to perform checkpoint proof"
         );
 
-        // Verify passed-in beaconStateRoot against oracle-provided block root:
-        BeaconChainProofs.verifyStateRootAgainstLatestBlockRoot({
-            latestBlockRoot: eigenPodManager.getBlockRootAtTimestamp(oracleTimestamp),
-            beaconStateRoot: stateRootProof.beaconStateRoot,
-            stateRootProof: stateRootProof.proof
+        Checkpoint memory checkpoint = _currentCheckpoint;
+
+        // Verify `balanceContainerProof` against `beaconBlockRoot`
+        BeaconChainProofs.verifyBalanceContainer({
+            beaconBlockRoot: checkpoint.beaconBlockRoot,
+            proof: balanceContainerProof
         });
 
-        VerifiedWithdrawal memory withdrawalSummary;
-        for (uint256 i = 0; i < withdrawalFields.length; i++) {
-            VerifiedWithdrawal memory verifiedWithdrawal = _verifyAndProcessWithdrawal(
-                stateRootProof.beaconStateRoot,
-                withdrawalProofs[i],
-                validatorFieldsProofs[i],
-                validatorFields[i],
-                withdrawalFields[i]
-            );
+        // Process each checkpoint proof submitted
+        uint64 exitedBalancesGwei;
+        for (uint256 i = 0; i < proofs.length; i++) {
+            ValidatorInfo memory validatorInfo = _validatorPubkeyHashToInfo[proofs[i].pubkeyHash];
```

Comment:
stuffing all of the validator specific checks into `_verifyCheckpointProof` cleaner imo since we write to it there

Reply:
yes but it's harder to get the `continue` behavior used in the loop here. any suggestions?

we could return a "skip" boolean but that's kinda messy

Reply:
[49769d4](https://github.com/Layr-Labs/eigenlayer-contracts/pull/515/commits/49769d436747913271e4f41d78d1f50ce9e398be)

I made this a little better by moving the actual state write into the parent method, making it more clear that there's a state update happening. I played with moving some of the checks/updates around and anything else felt messy.

Sidenote, for whatever reason the change in this commit saves 131 gas per validator :shrug: